### ISSUE TITLE
[Windows] Updater Auto Running As Admin

### ISSUE THREAD

**Issue by zQueal** [keybase/keybase-issues#2366]
I received this error randomly today.

![hoyiuh](https://cloud.githubusercontent.com/assets/239193/16779505/942731f2-4840-11e6-9e6d-224fa6866165.png)

```
[14C4:205C][2016-07-12T14:53:17]i001: Burn v3.10.2.2516, Windows v6.1 (Build 7601: Service Pack 1), path: C:\Users\zqueal\AppData\Local\Temp\{4D42DC84-6764-4436-BD62-DCB7B31FB734}\.cr\Keybase_1.0.16-20160712081028+3c54cc3.386.exe
[14C4:205C][2016-07-12T14:53:17]i000: Initializing string variable 'DokanProduct64' to value '{65A3A964-3DC3-0100-0000-160621082245}'
[14C4:205C][2016-07-12T14:53:17]i000: Initializing string variable 'DokanProduct86' to value '{65A3A986-3DC3-0100-0000-160621082245}'
[14C4:205C][2016-07-12T14:53:17]i009: Command Line: '-burn.clean.room=C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.VS7ZSBF2BD24WONVCBAORUI5HJ733HVTLEYLAPPIT66DDBIHWZIQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe'
[14C4:205C][2016-07-12T14:53:17]i000: Setting string variable 'WixBundleOriginalSource' to value 'C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.VS7ZSBF2BD24WONVCBAORUI5HJ733HVTLEYLAPPIT66DDBIHWZIQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe'
[14C4:205C][2016-07-12T14:53:17]i000: Setting string variable 'WixBundleOriginalSourceFolder' to value 'C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.VS7ZSBF2BD24WONVCBAORUI5HJ733HVTLEYLAPPIT66DDBIHWZIQ\'
[14C4:205C][2016-07-12T14:53:17]i000: Setting string variable 'LOGPATH_PROP' to value 'C:\Users\zqueal\AppData\Local\Temp\Keybase_20160712145317.log'
[14C4:205C][2016-07-12T14:53:17]i000: Setting string variable 'WixBundleName' to value 'Keybase'
[14C4:205C][2016-07-12T14:53:17]i000: Setting string variable 'WixBundleManufacturer' to value 'Keybase, Inc.'
[14C4:1CB8][2016-07-12T14:53:17]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
[14C4:1CB8][2016-07-12T14:53:17]i000: Setting version variable 'WixBundleFileVersion' to value '1.0.16.76'
[14C4:205C][2016-07-12T14:53:17]i100: Detect begin, 3 packages
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{65A3A986-3DC3-0100-0000-160621082245}'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT DokanUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{65A3A986-3DC3-0100-0000-160621082245}'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT DokanUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{65A3A964-3DC3-0100-0000-160621082245}'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT DokanUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Setting string variable 'DokanUninstallString' to value 'MsiExec.exe /X{65A3A964-3DC3-0100-0000-160621082245}'
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'Software\Microsoft\Windows\CurrentVersion\Uninstall\{70E747DE-4E09-44B0-ACAD-784AA9D79C02}_is1'
[14C4:205C][2016-07-12T14:53:17]i000: Setting numeric variable 'InnoCLIInstalled' to value 0
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'Software\Microsoft\Windows\CurrentVersion\Uninstall\{70E747DE-4E09-44B0-ACAD-784AA9D79C02}_is1'
[14C4:205C][2016-07-12T14:53:17]i000: Setting numeric variable 'InnoCLIInstalled64' to value 0
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{357F272E-BE0E-409F-8E39-0BB9827F5716}_is1'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT InnoUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{357F272E-BE0E-409F-8E39-0BB9827F5716}_is1'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT InnoUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{DEB2E54C-C39F-4DC8-93A7-ABE0AB91DDCA}_is1'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT InnoUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{DEB2E54C-C39F-4DC8-93A7-ABE0AB91DDCA}_is1'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT InnoUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{1B2672D9-2BAD-4C11-BA53-A75AF6FD7789}_is1'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT InnoUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry key not found. Key = 'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{1B2672D9-2BAD-4C11-BA53-A75AF6FD7789}_is1'
[14C4:205C][2016-07-12T14:53:17]i000: Registry value not found. Key = 'SOFTWARE\Keybase', Value = 'TargetDokanUninstallKey'
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT TargetDokanUninstallKey' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i000: Registry value not found. Key = 'SOFTWARE\Keybase', Value = 'TargetDokanUninstallKey'
[14C4:205C][2016-07-12T14:53:17]i000: Setting version variable 'WINTRUSTVERSION' to value '6.1.7601.18741'
[14C4:205C][2016-07-12T14:53:17]i102: Detected related bundle: {eae392ff-ba41-4374-b2a8-1acd75f3f4ee}, type: Upgrade, scope: PerUser, version: 1.0.16.69, operation: MajorUpgrade
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT InnoUninstallString' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'DokanUninstallString and NOT (WixBundleAction=3 AND ( TargetDokanUninstallKey <> DokanProduct64 AND TargetDokanUninstallKey <> DokanProduct86 ))' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i103: Detected related package: {1ED8ACE0-27AE-42B5-891A-E66DA256DFC0}, scope: PerUser, version: 1.0.16.69, language: 0 operation: MajorUpgrade
[14C4:205C][2016-07-12T14:53:17]i101: Detected package: runquiet.exe, state: Present, cached: None
[14C4:205C][2016-07-12T14:53:17]i101: Detected package: DokanSetup_redist.exe, state: Present, cached: Complete
[14C4:205C][2016-07-12T14:53:17]i101: Detected package: KeybasePrograms, state: Absent, cached: None
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'Installed OR VersionNT > v6.1 OR (VersionNT = v6.1 AND WINTRUSTVERSION >= v6.1.7601.18741)' evaluates to true.
[14C4:205C][2016-07-12T14:53:17]i052: Condition 'NOT WixBundleElevated' evaluates to false.
[14C4:205C][2016-07-12T14:53:17]e000: This installer must not be launched as administrator.
[14C4:205C][2016-07-12T14:53:17]e000: Error 0x81f40001: Bundle condition evaluated to false: NOT WixBundleElevated
[14C4:205C][2016-07-12T14:53:17]i199: Detect complete, result: 0x0
```

----------------------------

Manually updating via `upd check` also fails for the same reasons (picture):

```
2016/07/12 15:05:46 DEBG Using default path to keybase: C:\Users\zqueal\AppData\Roaming\Keybase\keybase
2016/07/12 15:05:46 DEBG Execute: C:\Users\zqueal\AppData\Roaming\Keybase\keybase [version -S]
2016/07/12 15:05:46 DEBG Execute: cmd [/c ver]
2016/07/12 15:05:46 INFO Checking for update, current version is 1.0.16-20160701140709+9c6469c
2016/07/12 15:05:46 INFO Using updater source: Keybase.io
2016/07/12 15:05:46 DEBG Using options: updater.UpdateOptions{Version:"1.0.16-20160701140709+9c6469c", Platform:"windows", DestinationPath:"C:\\Users\\zqueal\\AppData\\Roaming\\Keybase\\", URL:"", Channel:"", Env:"prod", Arch:"386", Force:false, OSVersion:"Microsoft Windows [Version 6.1.7601]", UpdaterVersion:"0.2.5"}
2016/07/12 15:05:46 INFO Request "https://api.keybase.io/_/api/1.0/pkg/update.json?auto_update=1&install_id=5f99522a73d6cda09d5cf4ab12a9cc0612b74315342f731d&os_version=Microsoft+Windows+%5BVersion+6.1.7601%5D&platform=windows&run_mode=prod&upd_version=0.2.5&version=1.0.16-20160701140709%2B9c6469c"
2016/07/12 15:05:46 DEBG Received update response: updater.Update{Version:"1.0.16-20160712081028+3c54cc3", Name:"v1.0.16-20160712081028+3c54cc3", Description:"Recent changes:\n---------------\n\n- Introducing the main GUI screen - this is where you'll look people up, manage your folders, and perform other actions. A lot of the features are stubbed out, but you can start playing with it.\n\n- Sharing before signup: go ahead and put data in /keybase/private/you,friend@twitter/. If you have any invite codes, this will pop up a window with a link to DM them. It also works for end-to-end encryption with Reddit, Coinbase, Github, and Hacker News users.\n\nWhat we are currently working on:\n---------------------------------\n\n- KBFS performance, including delayed writes\n", InstallID:"5f99522a73d6cda09d5cf4ab12a9cc0612b74315342f731d", RequestID:"e7af8d2f79f3b2a27fe64b1f", Type:0, PublishedAt:1468311028000, Asset:(*updater.Asset)(0x13242150), NeedUpdate:true}
2016/07/12 15:05:46 INFO Got update with version: 1.0.16-20160712081028+3c54cc3
2016/07/12 15:05:46 DEBG [Prompt for update]
2016/07/12 15:05:46 DEBG Auto update: true (set=true)
2016/07/12 15:05:46 INFO Reporting: https://api.keybase.io/_/api/1.0/pkg/act.json map[action:[auto] auto_update:[1] install_id:[5f99522a73d6cda09d5cf4ab12a9cc0612b74315342f731d] request_id:[e7af8d2f79f3b2a27fe64b1f] version:[1.0.16-20160701140709+9c6469c] upd_version:[0.2.5]]
2016/07/12 15:05:47 DEBG Creating: C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ

2016/07/12 15:05:47 INFO Request https://prerelease.keybase.io/windows/Keybase_1.0.16-20160712081028%2B3c54cc3.386.exe
2016/07/12 15:05:47 INFO Downloading to C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe.download
2016/07/12 15:06:02 INFO Downloaded 104808408 bytes
2016/07/12 15:06:03 INFO Verified digest: f48934de1bccb2778527ddfbdbb2f15d92f77799a8929316e5d06b660e55ad29 (C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe.download)
2016/07/12 15:06:03 INFO Moving C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe.download to C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe
2016/07/12 15:06:03 INFO Verify asset: C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe
2016/07/12 15:06:09 INFO Signed by deaa8ae7d06ea9aa49cc678ec49f2b1e1dddb63683e384db539a8649c47925f9
2016/07/12 15:06:09 DEBG Valid KID: deaa8ae7d06ea9aa49cc678ec49f2b1e1dddb63683e384db539a8649c47925f9
2016/07/12 15:06:09 INFO [Before apply]
2016/07/12 15:06:09 DEBG Execute: C:\Users\zqueal\AppData\Roaming\Keybase\keybase [update check-in-use]
2016/07/12 15:06:09 INFO [Applying update]
2016/07/12 15:06:09 DEBG Execute: C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe [/layout /quiet /log C:\Users\zqueal\AppData\Local\Temp\keybaseInstallLayout-NNXXB2CJCFQ5R5HLPH6FCQL4Q7FEKNKFH4SXILRVMZ44U4YCQYCQ]
2016/07/12 15:06:09 ERR  CheckCanBeSilent: Unable to execute C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe
2016/07/12 15:06:09 DEBG Execute: C:\Users\zqueal\AppData\Local\Temp\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ\Keybase_1.0.16-20160712081028+3c54cc3.386.exe []
2016/07/12 15:12:34 DEBG Remove temporary directory: "C:\\Users\\zqueal\\AppData\\Local\\Temp\\KeybaseUpdater.BLNBEWHJLGX7OXPPVYBR47MNW5XVNEU7UKN4I7ZUWHAFAQ4PF2MQ"
2016/07/12 15:12:34 INFO Reporting: https://api.keybase.io/_/api/1.0/pkg/error.json map[description:[Update Error (apply): exit status 1] install_id:[5f99522a73d6cda09d5cf4ab12a9cc0612b74315342f731d] request_id:[e7af8d2f79f3b2a27fe64b1f] version:[1.0.16-20160701140709+9c6469c] upd_version:[0.2.5] error_type:[apply]]
2016/07/12 15:12:34 ERR  [Update Error (apply): exit status 1]
```

**Comment by zanderz** [comment]
That was added on purpose because running the installer elevated like that means that the keybase.exe daemon is launched elevated, and can't be reached by other processes running as the regular user. It works after reboot, but confused people. Were you running everything elevated before?

**Comment by BrendanJ** [comment]
I'm getting the same exact issue.
My user account is an administrator with UAC disabled, so I'm not specifically running keybase with elevated privileges, but all attempted updates inevitably fail.

**Comment by zanderz** [comment]
Sorry guys, we'll revert that change, stay tuned.

**Comment by zQueal** [comment]
I don't believe I installed Keybase with elevated permissions, but my user account is an administrator. `¯\_(ツ)_/¯`

**Comment by zanderz** [comment]
Yesterday we released 1.0.16-20160713110422+a1c00b3 which reverts this restriction, so you should be able to update now. Please let us know if there are still problems.

**Comment by zQueal** [comment]
This version seems to have been installed and works perfectly! :+1:
